### PR TITLE
Allow more precise adjustment of Max Framerate option

### PR DESCRIPTION
**Pull request by Calinou**
### Proposed Changes

This allows catering to displays with variable refresh rate enabled. For example, when using a 120 Hz monitor, you can set the maximum framerate to 117, enable V-Sync and get low input lag without tearing. This works because the framerate remains just below the display's maximum VRR range, which prevents queued frames from accumulating.

Previously, you would either have to edit the settings file manually to achieve this, use a third-party FPS limiter (which is harder to set up and increases input lag), or stick to a lower FPS limit that is below the maximum VRR range (such as 110 FPS for 120 Hz).

The exact value to use in this scenario depends on your monitor refresh rate. Values to use for common refresh rates are:

- 60 Hz: 58 FPS
- 120 Hz: 117 FPS
- 144 Hz: 141 FPS
- 240 Hz: 235 FPS

For other refresh rates, you can calculate the FPS limit to use with `refresh_rate - (int(refresh_rate / 60.0) + 1)`.

See <https://blurbusters.com/howto-low-lag-vsync-on/> for more information.

### CODE REVIEW

**Comment by burgerindividual** [comment]
I don't think this will work due to the validation that Minecraft does when parsing the settings file. The reason the fps slider's increment was increased to 10 from 5 was for this reason, so it would match vanilla.

**Comment by Calinou** [comment]
> I don't think this will work due to the validation that Minecraft does when parsing the settings file. The reason the fps slider's increment was increased to 10 from 5 was for this reason, so it would match vanilla.

I've been wondering if vanilla could change this, but it would need some person with networking:tm: to be able to reach out to Mojang about this 🙂 (since it's technically not a bug)

**Comment by PweSol** [comment]
I've personally been using this for a long time https://github.com/CaffeineMC/sodium-fabric/issues/1156 and never had any noticeable issues.

After certain Sodium or Mc version, the FPS lock stopped saving in the menu but it still locks the actual game. This is probably related to the parsing of the settings file as mentioned by burgerguy.

However, there are still some more complications due to how Mc handles something else internally related to the frame rate (I forgot the details). We already had a more in-depth discussion about it in some other pull request or issue, but I can't find it.

tldr: this won't be merged / won't work until someone figures out and writes a better system of how Mc handles this

**Comment by IMS212** [comment]
This won't be merged, as this A: breaks vanilla compatibility and B: introduces many inconsistencies in how the FPS limiting works.

Vanilla has intentionally added this limit so users cannot break their settings; if Sodium allows invalid values, moving back to Vanilla will reset the settings and annoy users.
